raster/src/main/scala/geotrellis/raster/VectorToRaster.scala

```diff
@@ -41,14 +41,14 @@ object VectorToRaster {
     * @param      rasterExtent     Raster extent of the resulting raster.
     *
     * @note                        KernelDensity does not currently support Double raster data.
-    *                              If you use a Raster with a Double CellType (TypeFloat, TypeDouble)
+    *                              If you use a Raster with a Double CellType (FloatConstantNoDataCellType, DoubleConstantNoDataCellType)
     *                              the data values will be rounded to integers.
     */
   def kernelDensity[D](points: Seq[PointFeature[D]],
                        transform: D => Int, 
                        kernel: Kernel, 
                        rasterExtent: RasterExtent): Tile = {
-    val stamper = KernelStamper(TypeInt, rasterExtent.cols, rasterExtent.rows, kernel)
+    val stamper = KernelStamper(IntConstantNoDataCellType, rasterExtent.cols, rasterExtent.rows, kernel)
```

Comment:
This code doesn't deal with conversion of tiles to other tiles - rather, this code generates a kernel density using vector points and a raster extent. The alternative forms of celltype are really, primarily focused on geotiff interop - there's not an obvious argument to be made for extending this to Raw/UserDefinedNoData and not extending it to bytes, shorts, floats and doubles